Automattic/wp-calypso
# Domains Management i1: Contact button opens form which can start bulk domain update

Related to https://github.com/Automattic/dotcom-forge/issues/3433

## Proposed Changes

* Adds "Edit contact information" button to bulk actions toolbar
* Clicking button opens a contact details form with selected domains passed as query params
* The list of selected domains appears on the form so the user can remember what they're doing
* Submitting the form kicks off a bulk action
* After submitting the form the user is taken back to the domains table
* Refresh whois data when viewing domain contact details to ensure it reflects changes made by async jobs

The new page (defined in `bulk-edit-contact-info-page.tsx`) is mostly copy/pasted from `edit-contact-info-page/index.tsx`. I think   the copy/paste is appropriate because there are many parts of this page where we will want to slightly diverge from the existing contact form page I think. The actual form component itself (`<EditContactInfoFormCard>`) is shared, which is good.

I decided to pass the selected domains as an array query param. I'm not sure whether this is the best approach or not. Another idea was for the contact form to be more like a modal and not change the path, that way the selected domains would just be passed as a prop to the modal. But I decided against this because we want `<DomainsTable>` to be portable, and if someone embeds the domains table in some non-Calypso location then clicking the "Edit" button should navigate them into Calypso and they should be navigated back to their table again when done.

Things intentionally left out of this PR:
- Transfer lock toggle doesn't appear in the form when it should
- I believe we're missing some terms and conditions which sometimes appear at the bottom of the form
- Only works for the all-domains version of the table
- The "Apply contact update to My Account email" checkbox doesn't do anything and should probably be removed
- There's no breadcrumb at the top of the form page as shown in the mockups
- Different jurisdictions have different contact detail requirements. We should only be allowing bulk updates when all domains have compatible requirements. We're ignoring that for now

![CleanShot 2023-09-06 at 18 26 16@2x](https://github.com/Automattic/wp-calypso/assets/1500769/cbb5efd3-14fe-4c0f-b5b9-3d03ab8b0c48)

![CleanShot 2023-09-06 at 18 24 40@2x](https://github.com/Automattic/wp-calypso/assets/1500769/f4f2b646-f4f8-4280-afc9-b9db3c25aba7)

## Testing Instructions



- Make sure that either your sandbox has the latest code or you're not sandboxed
- `/domains/manage?flags=domains/management`
- Select domain(s)
- Click the "Edit contact information" button that appears
- Click
- The selected domains appear in the form sidebar
- Clicking cancel goes back to the domains table
- Clicking submit starts the bulk action and goes back to the domains table
- Check the domain(s) you updated, and you should see a `Spinner` saying there is an update pending 
- Next, log on to your sandbox and run `cat /tmp/php-errors` and run the command there `async-jobs/tester.php 87347463346 wpj_`
- On the front end, confirm you see a `Notice` saying the update was completed. Check the domain(s) you updated and the `Spinner` should have gone.

## Pre-merge Checklist
